src/explorer/assets/refresh.js

```diff
@@ -0,0 +1,25 @@
+// Mimi UI
+
+function refresh(speclist) {
+    
+    // Loop over the things that we want to show in the list, add
+    // one button for each element as a child to the variablelist div
+
+    for (var i in speclist) {
+
+        var newButton = document.createElement("button");
+        newButton.setAttribute("class", "tab");
+        
+        // Return a closure with a copy of the spec that's private to the function
+        newButton.onclick = (function() {
+            var spec = speclist[i]["VLspec"];
+
+            return function() {
+                vegaEmbed("#vis", spec, {actions: false});
+            }
+        }())
+
+        newButton.appendChild(document.createTextNode(speclist[i]["name"]));
+        document.getElementById("variablelist").appendChild(newButton);
```

Comment:
You could cache the return value of ``document.getElementById("variablelist")`` outside of the loop. Probably makes very little difference, but who knows, for very large model results it might help.